### PR TITLE
add support functions and session helpers

### PR DESCRIPTION
**Pull request by ericingram**
- Added `swell.functions` to handle app function calls
- Added `swell.session` to more easily work with session values
- Added tests and types

### CODE REVIEW

File: src/session.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      return request('get', '/session');
+    },
+
+    getCookie() {
+      this.state = opt.getCookie('swell-session');
+      return this.state;
+    },
+
+    setCookie(value) {
+      this.state = value;
```
Comment (swellmike) [inline src/session.js:15]: Maybe set `this.state` after setting the cookie

File: types/index.d.ts [hunk 2]
```diff
@@ -306,6 +306,34 @@ export namespace invoices {
...
+    functionName: string,
+    data?: any,
+    options?: any,
+  ): Promise<any>;
+  function post(
+    appId: string,
+    functionName: string,
+    data?: any,
+    options?: any,
+  ): Promise<any>;
```
Comment (swellmike) [inline types/index.d.ts:334]: delete?

File: src/session.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+function methods(request, opt) {
```
Comment (swellmike) [inline src/session.js:1]: Please add comments giving an overview of this module

File: src/functions.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+function methods(request, _opt) {
```
Comment (swellmike) [inline src/functions.js:1]: Please add comments giving an overview of this module

**Review by swellmike — CHANGES_REQUESTED** [review]
Made a review pass.  Just some code-style / documentation feedback.

File: types/index.d.ts (hunk 2 above)
Comment (ericingram) [reply types/index.d.ts:334]: Can't seem to use the word "delete" in a namespace like this. I tried a couple other approaches with no luck, then realized we also left `delete` out of the types for other namespaces

File: src/functions.js (hunk 4 above)
Comment (ericingram) [reply src/functions.js:1]: Added comments

File: src/session.js (hunk 3 above)
Comment (ericingram) [reply src/session.js:1]: Added comments

File: src/session.js (hunk 1 above)
Comment (ericingram) [reply src/session.js]: Removed the property, not really needed

**Review by swellmike — APPROVED** [review]

**Pull request merged**